dev/RELEASE_PROCESS.rst

```diff
@@ -3,40 +3,64 @@ Release Process
 
 This document describes the process for creating new releases.
 
-1. First, you should build wheels that you'd like to use for testing. That can
-   be done by following the `documentation for building wheels`_.
+1. **Increment the Python version:** Create a PR that increments the Python
+   package version. See `this example`_.
 
-2. **Testing:** Before a release is created, significant testing should be done.
+2. **Download the Travis-built wheels:** Once Travis has completed the tests,
+   the wheels from this commit can be downloaded from S3 to do testing, etc.
+   The URL is structured like this:
+   ``https://s3-us-west-2.amazonaws.com/ray-wheels/<hash>/ray-<version>-cp27-cp27mu-manylinux1_x86_64.whl``
+   where ``<hash>`` is replaced by the SHA of the commit and the ``<version>``
```

Comment:
Minor, but I prefer "commit ID" to "SHA" (occurs in 3 places). Up to you though.